packages/react/src/date-picker/DatePicker.tsx

```diff
@@ -200,20 +216,28 @@ export const DatePicker = ({
 				onBlur={onInputBlur}
 				onChange={onInputChange}
 				buttonRef={triggerRef}
-				buttonOnClick={toggleCalendar}
+				buttonOnClick={() => {
+					toggleCalendar();
+					setHasCalendarOpened(true);
+				}}
 				buttonAriaLabel={getDateInputButtonAriaLabel(inputValue)}
 			/>
 			<CalendarProvider yearRange={yearRange}>
-				{isCalendarOpen && (
-					<Popover {...popover.getPopoverProps()}>
-						<CalendarSingle
-							initialFocus
-							selected={valueAsDateOrUndefined}
-							onSelect={onSelect}
-							defaultMonth={defaultMonth}
-							numberOfMonths={1}
-							disabled={disabledCalendarDays}
-						/>
+				{/* We duplicate the Popover + Calendar as a workaround for a bug that scrolls the page to the top on initial open of the calandar - https://github.com/gpbl/react-day-picker/discussions/2059 */}
+				{hasCalendarOpened ? ( // If the calendar has opened at least once, we conditionally render the Popover + children to prevent the scroll-to-top bug
+					isCalendarOpen && (
+						<Popover {...popoverProps}>
+							<CalendarSingle {...calendarProps} />
+						</Popover>
+					)
+				) : (
+					// If the calendar has _not_ opened at least once, we conditionally render only the children of the Popover, i.e. the Calendar to prevent the UI jumping about everytime the calendar is opened
+					<Popover
+						{...popoverProps}
+						visibility={isCalendarOpen ? 'visible' : 'hidden'}
+						css={{ minHeight: '200px' }} // Using 200px as a safety buffer so that opening the date picker for the first time when the input is at the botom of the screen doesn't render the calendar virtually hidden, e.g. 2px height.
```

Comment:
```suggestion
						css={{ minHeight: '200px' }} // Using 200px as a safety buffer so that opening the date picker for the first time when the input is at the bottom of the screen can't render the calendar almost hidden, e.g. 2px height.
```

Reply:
Need to also do this for DateRangePicker